### PR TITLE
Pr l1tglobal emulator 80x

### PR DESCRIPTION
**Pull request by mulhearn**
Adds the 2016 L1 Global trigger (uGT) emulation and related data formats.

### CODE REVIEW

**Comment by mulhearn** [comment]
please test

**Comment by mulhearn** [comment]
+1

File: L1Trigger/L1TGlobal/src/L1TGlobalUtil.cc [hunk 1]
```diff
@@ -0,0 +1,529 @@
...
+// local cache & check on cacheIdentifier
+    unsigned long long l1GtMenuCacheID = evSetup.get<L1TGlobalTriggerMenuRcd>().cacheIdentifier();
+
+    if (m_l1GtMenuCacheID != l1GtMenuCacheID) {
+
+        //std::cout << "Attempting to get the Menu " << std::endl;
+        edm::ESHandle< TriggerMenu> l1GtMenu;
+        evSetup.get< L1TGlobalTriggerMenuRcd>().get(l1GtMenu) ;
+        m_l1GtMenu =  l1GtMenu.product();
+       //(const_cast<TriggerMenu*>(m_l1GtMenu))->buildGtConditionMap();
```
Comment (davidlange6) [inline L1Trigger/L1TGlobal/src/L1TGlobalUtil.cc]: @mulhearn m_l1GtMenu doesn't need to be a member data. (it is only consumed by m_algorithmMap).

But that asked, what is the cpu benefit of caching this?

File: L1Trigger/L1TGlobal/src/L1TGlobalUtil.cc [hunk 2]
```diff
@@ -0,0 +1,529 @@
...
+    prescale = (m_prescales.at(bit)).second;
+    return true;
+  }
+  
+  return false;  //couldn't get the information requested. 
+}
+const bool l1t::L1TGlobalUtil::getMaskByBit(int& bit, bool& mask) const {
+
+  // Need some check that this is a valid bit
+  if((m_masks.at(bit)).first != "NULL") {
```
Comment (davidlange6) [inline L1Trigger/L1TGlobal/src/L1TGlobalUtil.cc]: better to use [bit] instead of at(bit)

File: L1Trigger/L1TGlobal/src/L1TGlobalUtil.cc (hunk 1 above)
Comment (mulhearn) [inline L1Trigger/L1TGlobal/src/L1TGlobalUtil.cc]: This is the CondFormat that is short-lived... soon to be replaced by the one from UTM library instead.  You are asking how much we save wrt just extracting from the EventSetup during each event?  I don't think we tested.  What is the recommended practice (for next version)?

File: L1Trigger/L1TGlobal/src/L1TGlobalUtil.cc (hunk 2 above)
Comment (mulhearn) [inline L1Trigger/L1TGlobal/src/L1TGlobalUtil.cc]: will update PR

**Comment by mulhearn** [comment]
please test

**Comment by mulhearn** [comment]
+1

File: L1Trigger/L1TGlobal/src/L1TGlobalUtil.cc (hunk 1 above)
Comment (davidlange6) [inline L1Trigger/L1TGlobal/src/L1TGlobalUtil.cc]: recommended practice is not to cache unless it matters for performance

**Comment by davidlange6** [comment]
+1

**Pull request merged**